app/models/notification.rb

```diff
@@ -65,7 +65,10 @@ def previously_communicated_by?(method)
   end
 
   def preferred_communication_method
-    Flipper.enabled?(:whatsapp_appointment_reminders) ? "whatsapp" : nil
+    return "whatsapp" if Flipper.enabled?(:whatsapp_appointment_reminders)
+    imo_status = patient.imo_authorization&.status
+    return "imo" if Flipper.enabled?(:imo_messaging) && imo_status.in?(["invited", "subscribed"])
+    nil
```

Comment:
Food for thought - I see that this code assumes that both Whatsapp and IMO are never enabled together. If they were, this codepath would ignore IMO altogether. Should we allow `Notification` to be more flexible? Create a better communication method priority that can cascade from whatsapp to IMO to SMS, and detect which methods have already been attempted when choosing the next one?

I don't think this PR's the right place to _do_ that work, but wanted to run the idea by you here, as we now have _three_ available methods to contact patients :smile:

Reply:
Yeah, that's a fair point. I assumed the two would never be on together because that is the current plan and attempting to enforce that shouldn't be handled here. The only thing we have to do to make your suggestion work is determine which communication method is higher priority.

Reply:
Yup. Definitely not the right PR to address those concerns, but cool follow-up work/improvements for us to document before we release this work.